aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -69,12 +69,34 @@ pub async fn load(
     Ok(ProfileSet::parse(source)?)
 }
 
+type SectionType = String;
+type SectionName = String;
+// One example of what this looks like in practice:
+// ```
+// {
+//   "services": { // section_type
+//     "foo": { // section_name
+//       "ec2": { // sub_properties_name
+//          "endpoint_url": "https://foo.example" // sub-properties
+//        },
+//       "s3": { "endpoint_url": "https://foo.example", "some_other_property": "a value" },
+//     },
+//     "bar": {
+//       "ec2": { "endpoint_url": "https://bar.example" },
+//     }
+// }
+// ```
+// Dictionaries in Rust man... No fun at all.
+type OtherSections =
+    HashMap<SectionType, HashMap<SectionName, HashMap<String, HashMap<String, String>>>>;
```

Comment:
Should we adopt something similar to `serde_json::Value` or use our own `Document` type? Or better yet, perhaps we should make our own struct type for this use-case that just has lookup methods on it and uses the hash map nastiness internally?

Reply:
yeah we definitely need our own struct type here

Reply:
I don't think that Value or our own document is the right thing because this is stringy all the way down

Reply:
I created a struct.